### PR TITLE
OLMIS-598: Updated STYLE-GUIDE.md

### PR DESCRIPTION
**Pull request by lkukielka**
This change is [Reviewable](https://reviewable.io/reviews/openlmis/openlmis-template-service/9)

### CODE REVIEW

File: STYLE-GUIDE.md [hunk 1]
```diff
@@ -17,3 +17,179 @@ In most cases, the Hibernate DefaultNamingStrategy follows these conventions.  S
 * Table names should be pluralized.  This is to avoid *most* used words. e.g. orders instead of order
 * Table names with multiple words should be snake_case.
 * Column names with multiple words should be merged together.  e.g. `getFirstName()` would map to `firstname`
+
+# Unit tests 
+* __Every single test should be independent and isolated. Unit test shouldn't deped on another unit test.__ 
+
+  There is one incorrect example below:
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md]: I'd go with capslocked "DO:" / "DO NOT:"

File: STYLE-GUIDE.md [hunk 2]
```diff
@@ -17,3 +17,179 @@ In most cases, the Hibernate DefaultNamingStrategy follows these conventions.  S
...
+  ```java
+  @Test
+  public void testIsNotAnAdultIfAgeLessThan18() {
+    age = 17;
+    boolean isAdult = ageService.isAdult(age);
+    assertEquals(false, isAdult);
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md]: We should use assertTrue/assertFalse in all examples asserting flags

File: STYLE-GUIDE.md [hunk 3]
```diff
@@ -17,3 +17,179 @@ In most cases, the Hibernate DefaultNamingStrategy follows these conventions.  S
 * Table names should be pluralized.  This is to avoid *most* used words. e.g. orders instead of order
 * Table names with multiple words should be snake_case.
 * Column names with multiple words should be merged together.  e.g. `getFirstName()` would map to `firstname`
+
+# Unit tests 
+* __Every single test should be independent and isolated. Unit test shouldn't deped on another unit test.__
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md]: depend

File: STYLE-GUIDE.md [hunk 4]
```diff
@@ -17,3 +17,179 @@ In most cases, the Hibernate DefaultNamingStrategy follows these conventions.  S
...
+* __Every single test should be independent and isolated. Unit test shouldn't deped on another unit test.__ 
+
+  There is one incorrect example below:
+  ```java
+  @Test
+  public void testIsNotAnAdultIfAgeLessThan18() {
+    age = 17;
+    boolean isAdult = ageService.isAdult(age);
+    assertEquals(false, isAdult);
+    testIsNotAnAdultIfAgeMoreThan18();
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md]: What I understand with dependant tests is that the result of one depends on the execution of the other one. Not that they directly invoke each other. Eg. we keep a list of items and we add an item in the first test, asserting that the size is 1, then we add a new item in another test and assert that the size is 2 (which depends on whether the first one executed)

File: STYLE-GUIDE.md [hunk 5]
```diff
@@ -17,3 +17,179 @@ In most cases, the Hibernate DefaultNamingStrategy follows these conventions.  S
...
+    int age = 17;
+    boolean isAdult = ageService.isAdult(age);
+    assertEquals(false, isAdult);
+  }
+  ```
+* __Don't make unnecessary assertions. Don't assert mocked behavior, avoid assertions that check the exact same thing as another unit test.__
+
+  There is one incorrect example below:
+ ```java
+  @Test
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md:121]: We can also add verifying mocked result here, as a bad practice example, eg.

```
when(ageService.isAdult(18)).thenReturn(true);
assertTrue(ageService.isAdult());
```

File: STYLE-GUIDE.md [hunk 6]
```diff
@@ -17,3 +17,180 @@ In most cases, the Hibernate DefaultNamingStrategy follows these conventions.  S
...
+  public void testIsNotAnAdultIfAgeLessThan18() {
+    String uri = String.format("http://127.0.0.1:8080/age/", HOST, PORT);
+    HttpPost httpPost = new HttpPost(uri);
+    HttpResponse response = getHttpClient().execute(httpPost);
+    assertEquals(HttpStatus.ORDINAL_200_OK, response.getStatusLine().getStatusCode());
+  }
+  ```
+* __Unit test shouldn't test String Contexts. Integration test are better for this purpose.__
```
Comment (joshzamor) [inline STYLE-GUIDE.md]: Spring?

**Comment by pgesek** [comment]
@lkukielka conflicts need to be resolved

File: STYLE-GUIDE.md [hunk 7]
```diff
@@ -42,6 +42,183 @@ Examples:
...
+  }
+  ```
+  DO:
+    ```java
+
+  @Test
+  public void testIsNotAnAdultIfAgeLessThan18() {
+    int age = 17;
+    boolean isAdult = ageService.isAdult(age);
+    assertFalse(isAdult);
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md:87]: Missing closing curly bracket

File: STYLE-GUIDE.md [hunk 8]
```diff
@@ -42,6 +42,183 @@ Examples:
...
+
+  @Test
+  public void testIsPersonAbleToRunForPresident()
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md]: Missing opening curly bracket

File: STYLE-GUIDE.md [hunk 9]
```diff
@@ -42,6 +42,183 @@ Examples:
...
+ ```java
+  @Test
+  public void testIsNotAnAdultIfAgeLessThan18() {
+    String uri = String.format("http://127.0.0.1:8080/age/", HOST, PORT);
+    HttpPost httpPost = new HttpPost(uri);
+    HttpResponse response = getHttpClient().execute(httpPost);
+    assertEquals(HttpStatus.ORDINAL_200_OK, response.getStatusLine().getStatusCode());
+  }
+  ```
+* __Unit test shouldn't test Spring Contexts. Integration test are better for this purpose.__
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md]: tests are or test is

File: STYLE-GUIDE.md [hunk 10]
```diff
@@ -42,6 +42,183 @@ Examples:
...
+  @Test
+  public void testIsNotAnAdultIfAgeLessThan18() {
+    int age = 17;
+    assertEquals(17, age);
+
+    boolean isAdult = ageService.isAdult(age);
+    assertFalse(isAdult);
+  }
+  ```
+* __Unit test have to be independent from external resources (i.e. don't connect with databases or servers)__
```
Comment (sebbrudzinski) [inline STYLE-GUIDE.md]: has

**Pull request merged**